src/network/serverpackethandler.cpp

```diff
@@ -998,32 +998,43 @@ void Server::handleCommand_Interact(NetworkPacket *pkt)
 	verbosestream << "TOSERVER_INTERACT: action=" << (int)action << ", item="
 			<< item_i << ", pointed=" << pointed.dump() << std::endl;
 
-	RemotePlayer *player = m_env->getPlayer(pkt->getPeerId());
+	session_t peer_id = pkt->getPeerId();
+	RemotePlayer *player = m_env->getPlayer(peer_id);
 
 	if (player == NULL) {
 		errorstream << "Server::ProcessData(): Canceling: "
-				"No player for peer_id=" << pkt->getPeerId()
+				"No player for peer_id=" << peer_id
 				<< " disconnecting peer!" << std::endl;
-		DisconnectPeer(pkt->getPeerId());
+		DisconnectPeer(peer_id);
 		return;
 	}
 
 	PlayerSAO *playersao = player->getPlayerSAO();
 	if (playersao == NULL) {
 		errorstream << "Server::ProcessData(): Canceling: "
-				"No player object for peer_id=" << pkt->getPeerId()
+				"No player object for peer_id=" << peer_id
 				<< " disconnecting peer!" << std::endl;
-		DisconnectPeer(pkt->getPeerId());
+		DisconnectPeer(peer_id);
 		return;
 	}
 
+	// Get pointed to node, may be undefined
```

Comment:
and instead of letting this be undefined, why not keep it at the player_pos by default?

Reply:
I've moved this code.